### PR TITLE
Spike/investigate transactions

### PR DESCRIPTION
**Pull request by clewis1**
https://companieshouse.atlassian.net/browse/IDVA3-1128

Confirm a company and navigate to the psc type list screen.
Check the logs to confirm a transaction has been created.

`debug: Received transaction {"httpStatusCode":201,"resource":{"id":"181241-681217-115502","etag":"a7f287ad6fbf77a9570a7d1c1ae6a7351d19bef7","links":{"self":"/transactions/181241-681217-115502"},"reference":"PscVerificationReference","status":"open","kind":"transaction","createdAt":"2024-03-27T14:37:42Z","createdBy":{"language":"en","id":"Y2VkZWVlMzhlZWFjY2M4MzQ3MT","email":"demo@ch.gov.uk"},"updatedAt":"2024-03-27T14:37:42Z","description":"PSC Verification Transaction"}}`

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/services/internal/api.service.ts
Comment (PhilipGlover) [inline src/services/internal/api.service.ts:1]: do we need this service at all?

File: src/routers/confirmCompanyRouter.ts [hunk 1]
```diff
@@ -13,6 +15,14 @@ router.get("/", handleExceptions(async (req: Request, res: Response, _next: Next
 }));
 
 router.post("/", handleExceptions(async (req: Request, res: Response, _next: NextFunction) => {
+
+    const session = req.session as Session;
+    const companyNumber = req.query.companyNumber as string;
+    const DESCRIPTION = "PSC Verification Transaction";
+    const REFERENCE = "PscVerificationReference";
+
+    const transaction: Transaction = await postTransaction(session, companyNumber, DESCRIPTION, REFERENCE);
```
Comment (PhilipGlover) [inline src/routers/confirmCompanyRouter.ts:24]: in the companyProfile service, the request is passed to it, and then it pulls out the session and company number.  I wonder if we should be consistent, whichever way we go.
Plus something doesn't seem quite appropriate about passing the DESCRIPTION and REFERENCE.  I don't know the best place, but would just having them in the transaction service for now be neater?

**Review by PhilipGlover — CHANGES_REQUESTED** [review]
I’ve successfully run this code and seen tx created in db.  
Couple of issues above.

File: src/services/internal/transaction.service.ts
Comment (PhilipGlover) [inline src/services/internal/transaction.service.ts:1]: the internal / external split in services comes from the node starter project.  Whether we want to keep it I don't know.  But the transaction service is probably more external in the sense it's a public API, like the Company Profile.  The internal confirmCompany one is more providing functionality for this PSC Verify web.  Perhaps that should be a utility?  Perhaps we should scrap the internal / external split?

File: src/services/internal/api.service.ts
Comment (clewis1) [reply src/services/internal/api.service.ts:1]: I guess not now!

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by PhilipGlover — APPROVED** [review]
- changes addressed and have run the code again
- my comment re the external/internal services, can be considered late if needs be

**Review by hepsimo — APPROVED** [review]
Code looks good

File: src/services/internal/transaction.service.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    logger.debug(`Creating transaction with company number ${companyNumber}`);
+    const sdkResponse: Resource<Transaction> | ApiErrorResponse = await apiClient.transaction.postTransaction(transaction);
+
+    if (!sdkResponse) {
+        throw logger.error(`Transaction API POST request returned no response for company number ${companyNumber}`);
+    }
+
+    if (!sdkResponse.httpStatusCode || sdkResponse.httpStatusCode >= 400) {
+        throw logger.error(`Http status code ${sdkResponse.httpStatusCode} - Failed to post transaction for company number ${companyNumber}`);
+    }
```
[the remark this answers is not in the document]
Comment (PhilipGlover) [reply src/services/internal/transaction.service.ts:32]: I've added this question for discussion in our dev canvas

**Pull request merged**